### PR TITLE
Add documentation for Vector Collections in Jet [HZAI-77]

### PR DESCRIPTION
**Pull request by k-jamroz**
TODO:
- [ ] beta and enterprise markers
- [ ] link to [Vector collection data structure](https://github.com/hazelcast/hz-docs/pull/1124/) docs
- [ ] use consistent wording with V[ector collection data structure docs](https://github.com/hazelcast/hz-docs/pull/1124/)

### CODE REVIEW

**Review by rebekah-lawrence — CHANGES_REQUESTED** [review]

File: docs/modules/integrate/pages/file-connector.adoc [hunk 1]
```diff
@@ -631,6 +631,37 @@ This file format is not supported by SQL.
...
+BatchSourceMap.Entry<Integer, int[]> groundTruth = FileSources.files("/data")
+  .glob("groundtruth.ivecs")
+  .format(VectorSources.ivecsFormat())
+  .build();
+```
+--
+SQL::
++
+--
+This file format is not supported by SQL.
```
[the remark this answers is not in the document]
Comment (k-jamroz) [reply docs/modules/integrate/pages/file-connector.adoc:660]: I used the same text as for other formats unsupported in SQL. I can change all of them

File: docs/modules/integrate/pages/vector-collection-connector.adoc [hunk 2]
```diff
@@ -0,0 +1,56 @@
+= Vector Collection Connector
+[.enterprise]*{enterprise-product-name}*
+:page-beta: true
+:description: Vector collection stores vectors with metadata and allows efficient finding of the most similar entries based on vector distance.
+
```
[the remark this answers is not in the document]
Comment (k-jamroz) [reply docs/modules/integrate/pages/vector-collection-connector.adoc:5]: https://github.com/hazelcast/hz-docs/pull/1125/commits/13d4c1c4627a3f920705797ca22883d4c2497392

File: docs/modules/integrate/pages/vector-collection-connector.adoc [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+Vector collection stores vectors with metadata and allows efficient finding of the most similar entries based on vector distance.. See xref:data-structures:vector-collection.adoc[].
```
[the remark this answers is not in the document]
Comment (k-jamroz) [reply docs/modules/integrate/pages/vector-collection-connector.adoc]: https://github.com/hazelcast/hz-docs/pull/1125/commits/13d4c1c4627a3f920705797ca22883d4c2497392

File: docs/modules/integrate/pages/legacy-file-connector.adoc [hunk 4]
```diff
@@ -349,4 +349,22 @@ S3 sink uses the multi-part upload feature of S3 SDK. The sink buffers
...
 threshold. The multi-part upload is completed when the job completes and
 makes the objects available on the S3. Since a streaming jobs never
-complete, S3 sink is not currently applicable to streaming jobs.
\ No newline at end of file
+complete, S3 sink is not currently applicable to streaming jobs.
+
+=== Fvecs and ivecs
+[.enterprise]*{enterprise-product-name}*
+
+Fvecs and ivecs files are binary files containing respectively float and integer vectors with ids. The connectors are similar to the local file connectors, but work with fvecs and ivecs files.
```
[the remark this answers is not in the document]
Comment (k-jamroz) [reply docs/modules/integrate/pages/legacy-file-connector.adoc]: https://github.com/hazelcast/hz-docs/pull/1125/commits/13d4c1c4627a3f920705797ca22883d4c2497392

File: docs/modules/integrate/pages/file-connector.adoc [hunk 5]
```diff
@@ -631,6 +631,37 @@ This file format is not supported by SQL.
 --
 ====
 
+
+=== Fvecs and ivecs
+[.enterprise]*{enterprise-product-name}*
+
+Read binary files containing float and integer vectors with ids.
```
[the remark this answers is not in the document]
Comment (k-jamroz) [reply docs/modules/integrate/pages/file-connector.adoc]: https://github.com/hazelcast/hz-docs/pull/1125/commits/13d4c1c4627a3f920705797ca22883d4c2497392

**Comment by k-jamroz** [comment]
@rebekah-lawrence how can I add "beta" marker to a section of the page? `:page-beta:` works for entire page only

File: docs/modules/integrate/pages/file-connector.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (rebekah-lawrence) [reply docs/modules/integrate/pages/file-connector.adoc:660]: Don't give yourself extra work (if you haven't already!). This is one of the problems when reviewing just the changes - consistency is king; if it this way elsewhere, just leave it as is fornow and we'll catch it when we get round to working though the whole documentation to apply a style guideline.

**Comment by rebekah-lawrence** [comment]
@k-jamroz I cannot find any examples of Beta markers in sections. I have asked @oliverhowell if he knows whether there is a method we can use.

In the absence of a solution, I would suggest adding either (**BETA**) to the title (or immediately after it), or a warning that this is a Beta feature.

However, if there is time, wait to see if Oliver has a better solution!

File: docs/modules/integrate/pages/connectors.adoc [hunk 6]
```diff
@@ -346,4 +358,11 @@ The Jet API supports more connectors than SQL.
 |hazelcast-jet
 |streaming
 |at-least-once
+
+|xref:integrate:vector-collection-connector.adoc[VectorSinks.vectorCollection]
+[.enterprise]*{enterprise-product-name}*
```
Comment (EugeneAbramchuk) [inline docs/modules/integrate/pages/connectors.adoc:363]: `{enterprise-product-name}` is not resolving, is this intended?

File: docs/modules/integrate/pages/legacy-file-connector.adoc [hunk 7]
```diff
@@ -349,4 +349,24 @@ S3 sink uses the multi-part upload feature of S3 SDK. The sink buffers
...
\ No newline at end of file
+complete, S3 sink is not currently applicable to streaming jobs.
+
+== Fvecs and ivecs
+[.enterprise]*{enterprise-product-name}*
+
+Fvecs and ivecs files are binary files containing, respectively, float and integer vectors with IDs. The connectors are similar to local file connectors, but work with fvecs and ivecs files.
+Fvecs and ivecs files can only be sources, and can therefore only be read; fvecs and ivecs files cannot be used as sinks.
+
+```java
```
Comment (EugeneAbramchuk) [inline docs/modules/integrate/pages/legacy-file-connector.adoc:360]: do we have plans for Python samples too?

File: docs/modules/integrate/pages/vector-collection-connector.adoc [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+== Installing the Connector
+
+This connector is included in the full and slim {enterprise-product-name} distributions of Hazelcast.
+
+== Permissions
+If xref:security:enabling-jaas.adoc[security] is enabled, you can set up permissions to restrict clients' access to these data structures.
+
+To search in vector collection, you must add the `create` and `read` permissions for those collections. If you use the vector collection sink to write to vector collections, you must add the `create` and `put` permissions for those collections.
+
+For further information on adding these permissions, see xref:security:native-client-security.adoc[].
```
Comment (EugeneAbramchuk) [inline docs/modules/integrate/pages/vector-collection-connector.adoc:19]: the link looks broken and does not resolve

File: docs/modules/integrate/pages/connectors.adoc (hunk 6 above)
Comment (k-jamroz) [reply docs/modules/integrate/pages/connectors.adoc:363]: works for me in preview:

image

File: docs/modules/integrate/pages/legacy-file-connector.adoc (hunk 7 above)
Comment (k-jamroz) [reply docs/modules/integrate/pages/legacy-file-connector.adoc:360]: not here - Jet is only available in Java. 
AFAIK we did not implement fvecs/ivecs support in python, but there are some libraries that do that.

File: docs/modules/integrate/pages/vector-collection-connector.adoc (hunk 8 above)
Comment (k-jamroz) [reply docs/modules/integrate/pages/vector-collection-connector.adoc:19]: works for me in preview

**Review by oliverhowell — APPROVED** [review]

**Pull request merged**